### PR TITLE
GSoC final submission for Aniket Mathur

### LINKED ISSUE #67: GSoC final submission: Port to Python 3
@quozl @pro-panda please review the changes, I made in my personal branch.

https://github.com/Aniket21mathur/GSoC/blob/portToPython3/Ideas-2019.md#port-sugar-and-core-activities-to-python-3

### PR DESCRIPTION
**Pull request by pro-panda**
Fixes #67

### CODE REVIEW

File: Ideas-2019.md [hunk 1]
```diff
@@ -63,6 +63,91 @@ _(in chronological order)_
...
+ - https://github.com/sugarlabs/sugar/pull/844<br>
+
+**Port Sugar datastore to Python 3**<br>
+   Completed Python 3 port of Sugar datastore.<br>
+
+Link to pull request:<br>
+ - https://github.com/sugarlabs/sugar-datastore/pull/15<br>
+
+Link to packages:<br>
+ - https://github.com/Aniket21mathur/sugar-datastore-python3
```
Comment (pro-panda) [inline Ideas-2019.md]: Do not add links to personal repositories. We've often seen broken links due to this in the past

File: Ideas-2019.md [hunk 2]
```diff
@@ -63,6 +63,91 @@ _(in chronological order)_
...
+
+Link to pull requests:<br>
+ - https://github.com/sugarlabs/sin-dientes-activity/pull/18<br>
+ - https://github.com/sugarlabs/nutrition/pull/7<br>
+ - https://github.com/sugarlabs/collabwrapper/pull/18<br>
+ - https://github.com/sugarlabs/browse-activity/pull/97<br>
+ - https://github.com/sugarlabs/cedit-activity/pull/15<br>
+ - https://github.com/sugarlabs/clock-activity/pull/23<br>
+
+Open pull requests:<br>
```
Comment (pro-panda) [inline Ideas-2019.md]: PR status is dynamic and may change in future, to merged or closed. Any specific reason for adding a separate list of open PRs?

**Comment by quozl** [comment]
@Aniket21mathur, thanks for asking, we have been vague.  As you said in #67, the link you use has distracting pre-GSoC material before your "work review" section.  Move the "work review" section just below "selected", and use "work completed" as the heading.  Or add the work completed within the "project task checklist" section, expanding it as necessary.

Use links to commits, not links to pull requests.  The final work product is the merged chain of commits in a _master_ branch.  A pull request is a review that lasts only a short time, and the link to a pull request does not remain valid forever.  An open pull request is incomplete work, and it's your responsibility to get them reviewed and merged.

Work you completed is more than the merged commits.  You were [advised to keep a work diary](http://lists.sugarlabs.org/archive/sugar-devel/2019-May/056835.html) or engineering notebook at the start of GSoC; a working document that lists what you tried to do that did not necessarily result in a merged commit, but may have moved you forward.  Similar to @pro-panda's "areas where you faced issues and how you resolved them".  If it turns out you didn't keep the diary, I'm not surprised, as your difficulty in reporting daily or weekly progress really hinted at not keeping a diary.  If you have to reconstruct a diary, you might use your mail archive, pull requests, chat, stashes, and filesystem dates.

Google's [requirements](https://developers.google.com/open-source/gsoc/help/work-product) must be followed; 

> It must be easy to identify the work you have done. (i.e. the changes you made or new code.)
> When someone goes to the provided URL it should be clear what work you did without requiring them to do signifcant additional digging.

Links to merged commits are a minimum, but if the commit does not show the work you did (e.g. testing, abandoned prototypes, system configurations), you have to explain it.

> It should be in a stable location. The URL cannot be changed after submission.

Consider splitting your project out of _Ideas-2019.md_ and into a file _README.md_ in a directory _2019/work-product/port-to-python-3_.

After all your changes are complete, the link you use to the file in this repository may be qualified by a git hash (so that it cannot change), or you can have us add a static copy to people.sugarlabs.org.

> Someone else should be able to use the content at (or referenced from) the target of the link to extend your work.

Some of the work you did was not in the commits, so you should explain that.

> If your work is 100% complete, they should be able to use it.

That's why packaging sugar-toolkit-gtk3 was so critical.  At the moment, as far as I can see, I'm the only one who has been able to use your work.

> If your work is not 100% complete, it should be clear what's left to do.

Pull requests that haven't been merged could be listed here along with what needs to be done; and you may find that for some pull requests it might be easier to get something done than to say it has yet to be done.

Hope that helps.

@pro-panda, @nswarup14, please also review my comment here.

**Comment by nswarup14** [comment]
@quozl 
Reviewed the above and I agree with all the points. I shall open a similar PR, and keep the above points in mind.
Thanks.

**Comment by pro-panda** [comment]
Reviewed https://github.com/sugarlabs/GSoC/pull/68#issuecomment-524109545

**Comment by Aniket21mathur** [comment]
@quozl @pro-panda I made the changes suggested please review.

**Comment by quozl** [comment]
Reviewed.  See cb06b30630152011fcc62add02b35fbb3b89feae, b895de1665eb0fd6a415d91af03b2fd1a203953b, and 8c7b57fb53cbd2acfd86199b78aec48936289a95.

I think it would be good to mention your detailed investigation into how distribution packages are made, referencing the Debian bug, mailing list conversations, and so on.

Also that the project had many internal dependencies; you had to get the toolkit working before the datastore, and so on.

Also that the project was larger than the available time, but significant progress was made, resulting in working software.

**Comment by Aniket21mathur** [comment]
Reviewed the changes, thanks for the corrections.
 I will make the suggested changes very soon.

**Comment by Hrishi1999** [comment]
Off-topic, but should I do the same thing @Aniket21mathur is doing? Summarize my work to a readme file here? Or should I create a blog?

**Comment by Aniket21mathur** [comment]
@pro-panda @quozl can we merge this now? I have to complete the final evaluation, May I address the changes suggested by @quozl in a separate pr?

**Comment by Aniket21mathur** [comment]
Thanks!

**Pull request merged**